### DISCUSSION TITLE
Trying to remove all buttons from a message, getting "Cannot have empty row!"

### DISCUSSION THREAD

**Discussion by SvenWollinger** [discord-jda/JDA#2392] [Questions and Help] [answered]
Basically the title, im trying to edit a message in order to remove its buttons.
Currently im doing it like this:
```
message.editMessage(MessageEditData.fromMessage(message)).setActionRow().queue()
```

This throws ```Cannot have empty row!```

I researched a bit and found that using ```setActionRows``` should help, however that method doesnt exist on the newest version of JDA.

Would love to have some help on this, thanks!

**Comment by SvenWollinger** [comment, marked as answer]
I found a solution, but id still like to know if this is the intended way:

1. Loop over all components via message.components
2. Loop over any component that is an ActionRow
3. Remove said ActionRow if you want to clear the buttons
4. Re-apply via editMessageComponents()

Is this intentional? Surely it would be much simpler if setActionRow with an empty list would remove them, no?

**Reply by MinnDevelopment** [reply]
`setActionRow` is a shortcut for `setComponents(ActionRow.of(...))`. You are looking for `setComponents()` instead.

**Reply by SvenWollinger** [reply]
Got it, just wanted to know if this solution is okay to use. Thank you! :)

**Comment by MinnDevelopment** [comment]
You can use `message.editMessageComponents().queue()`
